Sources/Client.swift

```diff
@@ -277,6 +284,68 @@ extension Client {
     }
 
     /**
+     Fetch a collection of Assets from Contentful matching the specified query.
+
+     - Parameter query: The Query object to match results againts.
+     - Parameter completion: A handler being called on completion of the request.
+
+     - Returns: The data task being used, enables cancellation of requests.
+     */
+    @discardableResult public func fetchAssets(with query: AssetQuery,
+                                               then completion: @escaping ResultsHandler<ArrayResponse<Asset>>) -> URLSessionDataTask? {
+
+        let url = URL(forComponent: "assets", parameters: query.parameters)
+        return fetch(url: url, then: completion)
+    }
+
+    /**
+     Fetch a collection of Assets from Contentful matching the specified query.
+
+     - Parameter query: The Query object to match results againts.
+     - Returns: A tuple of data task and an observable for the resulting array of Assets.
+     */
+    @discardableResult public func fetchAssets(query: AssetQuery) -> Observable<Result<ArrayResponse<Asset>>> {
+        let asyncDataTask: AsyncDataTask<AssetQuery, ArrayResponse<Asset>> = fetchAssets(with:then:)
+        return toObservable(parameter: query, asyncDataTask: asyncDataTask).observable
+    }
+}
+
+
+// MARK: Mappable
+
+extension Client {
+    /**
+     Fetch a collection of Assets from Contentful matching the specified query.
+
+     - Parameter query: The Query object to match results againts.
+     - Returns: An Observable forr the resulting `MappedContent` container.
+     */
+    @discardableResult public func fetchMappedEntries(with query: Query) -> Observable<Result<MappedContent>> {
+        let asyncDataTask: AsyncDataTask<Query, MappedContent> = fetchMappedEntries(with:then:)
+        return toObservable(parameter: query, asyncDataTask: asyncDataTask).observable
+    }
+
+    /**
+     Fetches all entries and includes matching the passed in `Query`. The completion handler returned will return a `MappedContent` object which
+     contains an array of `Asset`s and a dictionary of ContentTypeId's to arrays of `EntryModellable` types of your own defining.
+     
+     - Parameter query: The Query object to match results against.
+     - Parameter completion: A handler being called on completion of the request containing a `MappedContent` instance.
+     
+     - Returns: The data task being used, enables cancellation of requests.
+     */
+    @discardableResult public func fetchMappedEntries(with query: Query,
+                                                      then completion: @escaping ResultsHandler<MappedContent>) -> URLSessionDataTask? {
+
+        let url = URL(forComponent: "entries", parameters: query.parameters)
+
+        return fetch(url: url) { (result: Result<ArrayResponse<Entry>>) in
+            let mappedResult: Result<MappedContent> = result.flatMap { return Result.success($0.toMappedContent(for: self.contentModel)) }
```

Comment:
Yes, in Swift land, `$0` is an example of  an "anonymous closure arguments" and generally i'm all for named arguments...I guess i was really hoping to squeeze it in one line, but i'm happy to split it up ;-)